docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -87,3 +87,18 @@ class C
     }
 }
 ```
+
+## Warn for redundant pattern in `not ... or <redundant>` pattern
+
+***Introduced in Visual Studio 2022 version 17.13***
+
+In a disjunctive `or` pattern such as `is not null or 42` or `is not int or string`
+the second pattern is redundant and likely results from misunderstanding the precedence order
+of `not` and `or` pattern combinators.  
+The compiler will provide a warning in such cases:
+```
+_ = o is not null or 42; // warning: pattern "42" is redundant
+_ = o is int or string; // warning: pattern "string" is redundant
+```
+It is likely that the user meant `is not (null or 42)` or `is not (int or string)` instead.
```

Comment:
Does this catch _all_ such patterns or just _most_ of them? As we discussed I think even catching _most_ of them, particularly the most common case that motivated this issue, would be a huge win. But I want to make sure that the documentation is clear on this point.

Reply:
I handled specific patterns, to be safer. But the PR handles all known patterns that were found in github searches.
I could refine to "The compiler provides a warning in common cases of this mistake"